File: src/main/java/br/com/augustoccesar/querybuilder/query/Limit.java
package br.com.augustoccesar.querybuilder.query;

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
public class Limit {
    private int value;

    public Limit(int value) {
        if (value < 0) {
            throw new IllegalArgumentException("Limit value must be positive.");
        } else {
            this.value = value;
        }
    }

    public int getValue() {
        return value;
    }
}

File: src/main/java/br/com/augustoccesar/querybuilder/query/Limit.kt
package br.com.augustoccesar.querybuilder.query

/**
 * Author: augustoccesar
 * Date: 10/05/17
 */
class Limit(value: Int) {
    var value: Int = 0
        private set

    init {
        if (value < 0) {
            throw IllegalArgumentException("Limit value must be positive.")
        } else {
            this.value = value
        }
    }
}
